homeassistant/components/landisgyr_heat_meter/const.py

```diff
@@ -0,0 +1,192 @@
+"""Constants for the Landis+Gyr Heat Meter integration."""
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import ENERGY_MEGA_WATT_HOUR, TEMP_CELSIUS, VOLUME_CUBIC_METERS
+from homeassistant.helpers.entity import EntityCategory
+
+DOMAIN = "landisgyr_heat_meter"
+
+GJ_TO_MWH = 0.277778  # conversion factor
+
+HEAT_METER_SENSOR_TYPES = (
+    SensorEntityDescription(
+        key="heat_usage",
+        icon="mdi:fire",
+        name="Heat usage",
+        native_unit_of_measurement=ENERGY_MEGA_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        key="volume_usage_m3",
+        icon="mdi:fire",
+        name="Volume usage",
+        native_unit_of_measurement=VOLUME_CUBIC_METERS,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    # Diagnostic entity for debugging, this will match the value in GJ indicated on the meter's display
+    SensorEntityDescription(
+        key="heat_usage_gj",
+        icon="mdi:fire",
+        name="Heat usage GJ",
+        native_unit_of_measurement="GJ",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="heat_previous_year",
+        icon="mdi:fire",
+        name="Heat usage previous year",
+        native_unit_of_measurement=ENERGY_MEGA_WATT_HOUR,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="volume_previous_year_m3",
+        icon="mdi:fire",
+        name="Volume usage previous year",
+        native_unit_of_measurement=VOLUME_CUBIC_METERS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="ownership_number",
+        name="Ownership number",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="error_number",
+        name="Error number",
+        icon="mdi:home-alert",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="device_number",
+        name="Device number",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="measurement_period_minutes",
+        name="Measurement period minutes",
```

Comment:
Is this a duration measured in minutes? If so, we should set that unit of measurement and use the duration device class.